### ISSUE TITLE
GPU OOM issue

### ISSUE THREAD

**Issue by DrChiZhang** [kwsong0113/diffusion-forcing-transformer#33]
Hi, Thank you for the amazing work!
I have tried to train this model with 8 * A100 with 80GB VRAM and it encounters OOM error even with very small batch size 2.
Any clue for this issue ?
thanks

CONFIG: 

experiment:
  debug: ${debug}
  tasks:
  - training
  num_nodes: 1
  training:
    precision: 16-mixed
    compile: false
    lr: 8.0e-06
    batch_size: 2
    max_epochs: 24
    max_steps: -1
    max_time: null
    data:
      num_workers: 11
      shuffle: false
    optim:
      accumulate_grad_batches: 1
      gradient_clip_val: 1.0
    checkpointing:
      every_n_train_steps: null
      every_n_epochs: 1
      train_time_interval: null
      enable_version_counter: false
      save_top_k: -1
  validation:
    precision: 16-mixed
    compile: false
    batch_size: 8
    val_every_n_step: 1.0
    val_every_n_epoch: 1
    num_sanity_val_steps: 1
    limit_batch: 1.0
    data:
      num_workers: 0
      shuffle: false
    inference_mode: true
  test:
    precision: 16-mixed
    compile: false
    batch_size: 4
    limit_batch: 1.0
    data:
      num_workers: 0
      shuffle: false
    inference_mode: true
  find_unused_parameters: false
  reload_dataloaders_every_n_epochs: 1
  ema:
    enable: true
    decay: 0.9999
    validate_original_weights: false
dataset:
  debug: ${debug}
  save_dir: data/real-estate-10k
  data_mean:
  - - - 0.577
  - - - 0.517
  - - - 0.461
  data_std:
  - - - 0.249
  - - - 0.249
  - - - 0.268
  latent:
    enable: false
    type: pre_sample
    suffix: null
    downsampling_factor:
    - 1
    - 8
    num_channels: 4
  resolution: 256
  observation_shape:
  - 3
  - ${dataset.resolution}
  - ${dataset.resolution}
  max_frames: 16
  n_frames: ${dataset.max_frames}
  context_length: 4
  frame_skip: 10
  filter_min_len: null
  external_cond_dim: 16
  external_cond_stack: false
  external_cond_processing: null
  preload: false
  subdataset_size: 10000
  num_eval_videos: 20
  maximize_training_data: true
  augmentation:
    frame_skip_increase: 1
    horizontal_flip_prob: 0.5
    reverse_prob: 0.5
    back_and_forth_prob: 0.1
algorithm:
  debug: ${debug}
  lr: ${experiment.training.lr}
  backbone:
    name: dit3d_pose
    variant: full
    pos_emb_type: rope_3d
    patch_size: 2
    hidden_size: 384
    depth: 12
    num_heads: 6
    mlp_ratio: 4.0
    use_gradient_checkpointing: false
    conditioning:
      modeling: film
      type: ${algorithm.camera_pose_conditioning.type}
      dim: null
    external_cond_dropout: 0.1
    channels:
    - 128
    - 256
    - 576
    - 1152
    num_updown_blocks:
    - 3
    - 3
    - 6
    num_mid_blocks: 20
    use_checkpointing:
    - false
    - false
    - false
    - true
    use_fourier_noise_embedding: true
  x_shape: ${dataset.observation_shape}
  max_frames: ${dataset.max_frames}
  n_frames: ${dataset.n_frames}
  frame_skip: ${dataset.frame_skip}
  context_frames: ${dataset.context_length}
  latent: ${dataset.latent}
  data_mean: ${dataset.data_mean}
  data_std: ${dataset.data_std}
  external_cond_dim: ${dataset.external_cond_dim}
  external_cond_stack: ${dataset.external_cond_stack}
  external_cond_processing: ${dataset.external_cond_processing}
  compile: true_without_ddp_optimizer
  weight_decay: 0.01
  optimizer_beta:
  - 0.9
  - 0.99
  lr_scheduler:
    name: constant_with_warmup
    num_warmup_steps: 10000
    num_training_steps: 550000
  noise_level: random_independent
  uniform_future:
    enabled: false
  fixed_context:
    enabled: false
    indices: null
    dropout: 0
  variable_context:
    enabled: false
    prob: 0
    dropout: 0
  chunk_size: -1
  scheduling_matrix: full_sequence
  replacement: noisy_scale
  diffusion:
    is_continuous: true
    timesteps: 1000
    beta_schedule: cosine_simple_diffusion
    schedule_fn_kwargs:
      shift: 1.0
      shifted: 0.125
      interpolated: false
    use_causal_mask: false
    clip_noise: 20.0
    objective: pred_v
    loss_weighting:
      strategy: sigmoid
      snr_clip: 5.0
      cum_snr_decay: 0.9
      sigmoid_bias: -1.0
    sampling_timesteps: 50
    ddim_sampling_eta: 0.0
    reconstruction_guidance: 0.0
    training_schedule:
      name: cosine
      shift: 0.125
    precond_scale: 0.125
  vae:
    pretrained_path: null
    pretrained_kwargs: {}
    use_fp16: true
    batch_size: 2
  checkpoint:
    reset_optimizer: false
    strict: true
  tasks:
    prediction:
      enabled: true
      history_guidance:
        name: conditional
      keyframe_density: null
      sliding_context_len: null
    interpolation:
      enabled: false
      history_guidance:
        name: conditional
      max_batch_size: null
  logging:
    deterministic: 0
    loss_freq: 100
    grad_norm_freq: 100
    max_num_videos: 256
    n_metrics_frames: null
    metrics:
    - fvd
    - is
    - fid
    - lpips
    - mse
    - ssim
    - psnr
    metrics_batch_size: 16
    sanity_generation: false
    raw_dir: null
  camera_pose_conditioning:
    normalize_by: first
    bound: null
    type: ray_encoding
debug: false
wandb:
  entity: zhangchi0118-huawei
  project: dfot
  mode: offline
resume: null
load: null
name: DFoT_RE10k

**Comment by DrChiZhang** [comment]
BTW, set batch size =1 seems works.
